src/stores/AppStore.js

```diff
@@ -136,8 +136,16 @@ export default types
     userLabels: isFF(FF_DEV_1536) ? types.optional(UserLabels, { controls: {} }) : types.undefined,
   })
   .preProcessSnapshot((sn) => {
-    if (!sn.user) {
-      sn.user = window.APP_SETTINGS?.user ?? null;
+    if (isFF(FF_DEV_2887)) {
+      const currentUser = window.APP_SETTINGS?.user ?? null;
+
+      if (sn.users?.length) {
+        sn.users = [currentUser, ...sn.users.filter(({ id }) => id !== currentUser.id)];
+      }
+
+      if (currentUser?.id) {
+        sn.user = currentUser.id;
```

Comment:
we should always set `sn.user` otherwise it can break reference if user came from constructor

Reply:
I agree, see my other comment.